lib/libthr/thread/thr_exit.c

```diff
@@ -322,7 +326,11 @@ exit_thread(void)
 	 * Kernel will do wakeup at the address, so joiner thread
 	 * will be resumed if it is sleeping at the address.
 	 */
+#if defined(__CHERI_PURE_CAPABILITY__) && defined(RTLD_SANDBOX)
+	_rtld_thr_exit(&curthread->tid);
```

Comment:
The thing is that the final `thr_exit` syscall must be made from the linker because the linker needs to clean up all the compartment's stacks before terminating the thread. So libthr has to hand control to the linker.

Reply:
So why not have the thr_exit libc wrapper call into rtld instead then?

Reply:
It looks like thr_exit does not have a libc wrapper and goes straight to assembly...

Reply:
I pushed this [e3c8e39](https://github.com/CTSRD-CHERI/cheribsd/pull/1653/commits/e3c8e39fe6c3684bafd0ed7f1fe850757ed198d5) which implements redirecting the thr_exit wrapper to call _rtld_thr_exit instead.